### PR TITLE
feat(Vinnueftirlitid new template): change machine supervisor

### PR DESCRIPTION
**Pull request by davidkef**
# ...

Attach a link to issue if relevant

## What

Specify what you're trying to achieve

## Why

Specify why you need to achieve this

## Screenshots / Gifs

Attach Screenshots / Gifs to help reviewers understand the scope of the pull request

## Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Formatting passes locally with my changes
- [ ] I have rebased against main before asking for a review

### CODE REVIEW

**Review by oddsson — APPROVED** [review]

**Review by thorkellmani — APPROVED** [review]
LGTM

**Review by berglindoma13 — CHANGES_REQUESTED** [review]
*Vantar validation á seinasta skref, kemur í console en koma engin villuskilaboð á skjáinn

File: libs/application/templates/aosh/change-machine-supervisor/src/forms/ChangeMachineSupervisorForm/InformationSection/supervisorSubSection.ts [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+      id: 'supervisorMultiField',
+      title: supervisor.general.newSupervisorTitle,
+      description: supervisor.general.description,
+      children: [
+        buildCheckboxField({
+          id: 'supervisor.isOwner',
+          title: supervisor.general.title,
+          width: 'full',
+          options: [
+            { value: 'ownerIsSupervisor', label: 'Eigandi er umráðamaður' },
```
Comment (berglindoma13) [inline libs/application/templates/aosh/change-machine-supervisor/src/forms/ChangeMachineSupervisorForm/InformationSection/supervisorSubSection.ts:30]: Er þetta ekki þýðanlegt yfir á ensku?

File: libs/application/templates/aosh/change-machine-supervisor/src/lib/messages/application.ts [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    description:
+      'Description of application state/status when application is done',
+  },
+  historyLogApprovedByReviewer: {
+    id: 'aosh.cms.application:historyLogApprovedByReviewer',
+    defaultMessage: 'Samþykkt af öllum aðilum',
+    description: 'History log approved by reviewer',
+  },
+  actionCardPrerequisites: {
+    id: 'hlc.application:actionCardPrerequisites',
```
Comment (berglindoma13) [inline libs/application/templates/aosh/change-machine-supervisor/src/lib/messages/application.ts:44]: þarf að breyta úr hlc í aosh

File: libs/application/templates/aosh/change-machine-supervisor/src/lib/messages/information.ts [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+        defaultMessage: 'Fjarlægja umráðamann',
+        description: 'Operator remove label',
+      },
+      add: {
+        id: 'aosh.cms.application:information.labels.supervisor.add',
+        defaultMessage: 'Bæta við umráðamanni',
+        description: 'Operator add label',
+      },
+      identicalError: {
+        id: 'ta.cms.application:information.labels.supervisor.identicalError',
```
Comment (berglindoma13) [inline libs/application/templates/aosh/change-machine-supervisor/src/lib/messages/information.ts:202]: þarf að breyta út ta í aosh

**Review by berglindoma13 — CHANGES_REQUESTED** [review]

File: libs/application/templates/aosh/change-machine-supervisor/src/forms/ChangeMachineSupervisorForm/InformationSection/supervisorSubSection.ts (hunk 1 above)
Comment (davidkef) [reply libs/application/templates/aosh/change-machine-supervisor/src/forms/ChangeMachineSupervisorForm/InformationSection/supervisorSubSection.ts:30]: Jú, ætti ég ekki að setja þetta bara í messages þar sem við getum verið með íslenska og enska þýðingu fyrir þetta

File: libs/application/templates/aosh/change-machine-supervisor/src/lib/messages/application.ts (hunk 2 above)
Comment (davidkef) [reply libs/application/templates/aosh/change-machine-supervisor/src/lib/messages/application.ts:44]: fixed

File: libs/application/templates/aosh/change-machine-supervisor/src/lib/messages/information.ts (hunk 3 above)
Comment (davidkef) [reply libs/application/templates/aosh/change-machine-supervisor/src/lib/messages/information.ts:202]: Fixed

File: libs/application/templates/aosh/change-machine-supervisor/src/forms/ChangeMachineSupervisorForm/InformationSection/supervisorSubSection.ts (hunk 1 above)
Comment (davidkef) [reply libs/application/templates/aosh/change-machine-supervisor/src/forms/ChangeMachineSupervisorForm/InformationSection/supervisorSubSection.ts:30]: Fixed

**Review by saevarma — APPROVED** [review]
Core file LGTM

**Review by berglindoma13 — APPROVED** [review]
LGTM

File: libs/application/templates/aosh/change-machine-supervisor/src/fields/MachinesField/MachineSelectField.tsx [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+import { useLazyMachineDetails } from '../../hooks/useLazyMachineDetails'
+import { useFormContext } from 'react-hook-form'
+import { getValueViaPath } from '@island.is/application/core'
+import { MachineDto } from '@island.is/clients/work-machines'
+
+interface MachineSearchFieldProps {
+  currentMachineList: MachineDto[]
+}
+
+export const MachineSelectField: FC<
```
[the remark this answers is not in the document]
Comment (davidkef) [reply libs/application/templates/aosh/change-machine-supervisor/src/fields/MachinesField/MachineSelectField.tsx:24]: Are you referring to the one coming from the Transfer-of-machine-ownership application? Which it is.
But I think that would be another task to rewrite that so it's coming from a shared place, same goes with Samgöngustofu. That would also need to be coming from a shared place if we are going that route.
Or are you talking about it being the same as the one in pickMachineSubSection, which it's not

**Review by Thorrik58 — APPROVED** [review]

File: libs/application/templates/aosh/change-machine-supervisor/src/fields/MachinesField/MachineSelectField.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (davidkef) [reply libs/application/templates/aosh/change-machine-supervisor/src/fields/MachinesField/MachineSelectField.tsx:24]: Yes I agree with you. Thank you, it'll be added to the our backlog as a task

**Pull request merged**